Content.Server/Shuttle/ShuttleDestinationSlotSystem.cs

```diff
@@ -0,0 +1,163 @@
+using Content.Shared.Containers.ItemSlots;
+using Content.Server.Shuttle.Components;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Containers;
+using Robust.Shared.Map;
+using Robust.Shared.Player;
+using Robust.Shared.Random;
+using Content.Server.Station.Systems;
+using Content.Server.Shuttles.Components;
+using Content.Server.Shuttles.Systems;
+
+namespace Content.Server.Shuttle;
+
+public sealed class ShuttleDestinationSlotSystem : EntitySystem
+{
+    [Dependency] private readonly ItemSlotsSystem _itemSlots = default!;
+    [Dependency] private readonly ShuttleConsoleSystem _console = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ShuttleDestinationSlotComponent, ComponentInit>(OnInit);
```

Comment:
Because you're creating components in `OnInit()`, this is causing `UninitializedSaveTest` to fail, which may potentially result in funky behavior when maps containing shuttle consoles are saved.

It'd be more ideal to instead opt for `MapInitEvent` here, which will initialize the component and create the item slot when the map initializes, rather than as soon as the component merely exists in some capacity. This will also fix the unit test that this PR is currently failing

Reply:
Thanks; I was using the Nuke as a starting point as it had the item slot functionality already, so this was something that slipped by.
I'll get a fix in. 👍